### PR TITLE
Fix imaginator to fail if umask!=022.

### PR DESCRIPTION
**Pull request by rgooch**
This relates to issue #445.

### CODE REVIEW

File: cmd/imaginator/main.go [hunk 1]
```diff
@@ -47,6 +47,10 @@ func main() {
 		os.Exit(1)
 	}
 	logger := serverlogger.New("")
+	if umask := syscall.Umask(022); umask != 022 {
```
Comment (masiulaniec) [inline cmd/imaginator/main.go]: `syscall.Umask` never fails but it returns the previous value. Why does it matter for imaginator what the old value was?

**Comment by rgooch** [comment]
See the comment on the next line: since you can't cleanly and reliably _set_ the umask in Go, it's simpler to fail if the umask isn't 022.

**Comment by masiulaniec** [comment]
Indeed, I encountered this property of Linux before when trying to add `sshd`-style sandboxing to some Go software: `setresuid` doesn't have process-wide effect on Linux. Personally, whenever I get curious about Unix implementation details, my go-to kernel to study is the one from OpenBSD; I guess that makes me a member of a diminishing group :-)

I still don't understand the diff. If `umask`'s immediate effect is only on the calling thread, then don't we also need something like a surrounding `runtime.LockOSThread`/`runtime.UnlockOSThread` to make sure that the newly set umask gets used for our file operations? Go does not guarantee any goroutine-to-thread affinity without these calls. I'm not seeing any relevant `runtime.LockOSThread` in cmd/imaginator.

**Comment by rgooch** [comment]
Again, the call to `syscall.Umask` is to _check_ the umask (and then later fail if umask != 022). Since there isn't a call to _only_ check the umask, I have to pass in the desired value (which will change nothing if the umask is already 022).

**Pull request merged**